stream-chat-android-ui-common/src/main/kotlin/io/getstream/chat/android/ui/common/feature/messages/list/MessageListController.kt

```diff
@@ -1274,14 +1426,33 @@ public class MessageListController(
      *
      * @return The [Message] with the given id or null if the message is not in the list.
      */
-    public fun getMessageById(messageId: String): Message? {
+    public fun getMessageFromListStateById(messageId: String): Message? {
         return (
-            _messageListState.value.messageItems.firstOrNull {
+            listState.value.messageItems.firstOrNull {
                 it is MessageItemState && it.message.id == messageId
             } as? MessageItemState
             )?.message
     }
 
+    /**
+     * Fetches the message with the given ID. Spares unnecessary API calls by first trying to load
+     * the message from the database and only if that fails will it then initiate an API call.
+     *
+     * @param messageId The ID of the message we are fetching.
+     *
+     * @return The message with the given ID if it was successfully fetched, null otherwise.
+     */
+    public suspend fun getMessageById(messageId: String): Message? {
+        val result = chatClient.getMessageUsingCache(messageId).await()
+
+        return when (result) {
+            is Result.Success -> result.value
+            is Result.Failure -> {
+                null
+            }
+        }
+    }
+
```

Comment:
Get message by id is not used anywhere and the Compose/XML VMs use `getMessageFromListStateById` for their `getMEssageById` function. This creates an API mismatch of sorts. Do we need these two functions or can it be handled just by `getMessageById`?